File: credit/credit.c
#include <stdio.h>
#include <cs50.h>

int length(int s);

int main(void){

    int sum = 0;
    int x;
    int n = get_long("Number: ");
    int l = length(n);

    for(int i = 0; i < l / 2; i ++){

        sum  = sum + n % 10;
        n = n / 10;
        x = n % 10 * 2;
        n = n / 10;
        sum = sum + x / 10 + x % 10;

    }

    if(sum % 10 == 0){

        if(l == 13 || l == 16)
            printf("VISA\n");

        else if(l == 15)
            printf("AMEX\n");

        else
            printf("MASTERCARD\n");

    }

    else
        printf("INVALID\n");

}

int length(int s){
    int length = 1;
    while(s / 10 > 0){
        s = s / 10;
        length ++;
    }

return length;
}

File: sentimental-credit/credit.py
from cs50 import get_string

def main():
    n = get_string("Number: ")



print(sum)
if sum % 10 == 0:
    if length == 15 and n[0] + n[1] in [34, 37]:
        print("AMEX")
    elif length == 16 and n[0] + n[1] in [51, 52, 53, 54, 55]:
        print("MASTERCARD")
    elif length == 13 or length == 16 and n[0] == 4:
        print("VISA")

else:
    print("INVALID")


def isvalid():
    length = len(n)
    sum = 0

    for i in range(length):
        m = int(n[i])
        if i % 2 == 0:

            if  m * 2 >= 10:
                sum += m * 2 // 10 + m * 2% 10
            else:
                sum += m * 2

        else:
            sum += m
